dialect/sql/builder_test.go

```diff
@@ -1711,22 +1711,23 @@ func TestInsert_OnConflict(t *testing.T) {
 	t.Run("Postgres", func(t *testing.T) { // And SQLite.
 		query, args := Dialect(dialect.Postgres).
 			Insert("users").
-			Columns("id", "email").
-			Values("1", "user@example.com").
+			Columns("id", "email", "creation_time").
+			Values("1", "user@example.com", 1633279231).
 			OnConflict(
 				ConflictColumns("email"),
 				ConflictWhere(EQ("name", "Ariel")),
 				ResolveWithNewValues(),
 				// Update all new values excepts id field.
 				ResolveWith(func(u *UpdateSet) {
 					u.SetIgnore("id")
+					u.SetIgnore("creation_time")
 					u.Add("version", 1)
 				}),
 				UpdateWhere(NEQ("updated_at", 0)),
 			).
 			Query()
-		require.Equal(t, `INSERT INTO "users" ("id", "email") VALUES ($1, $2) ON CONFLICT ("email") WHERE "name" = $3 DO UPDATE SET "id" = "users"."id", "email" = "excluded"."email", "version" = COALESCE("users"."version", 0) + $4 WHERE "updated_at" <> $5`, query)
-		require.Equal(t, []interface{}{"1", "user@example.com", "Ariel", 1, 0}, args)
+		require.Equal(t, `INSERT INTO "users" ("id", "email", "creation_time") VALUES ($1, $2, $3) ON CONFLICT ("email") WHERE "name" = $4 DO UPDATE SET "id" = "users"."id", "email" = "excluded"."email", "creation_time" = "users"."creation_time", "version" = COALESCE("users"."version", 0) + $5 WHERE "updated_at" <> $6`, query)
```

Comment:
Just out of curiosity, why include `"id" = "users"."id"` and `"creation_time" = "users"."creation_time"` instead of omitting them? I'm assuming it's more work to remove?

Reply:
Note that, this is the internal package for building the SQL statement and not the generated code. But for your question, I prefer to use explicit ignore (assign to database values), because in the case the create command contains only immutable fields, omitting them will make the UPSERT statement invalid and the ON CONFLICT clause will need to be removed. This approach just simplifies the implementation and also triggers update hooks in the database.